runtime/lua/vim/diagnostic.lua

```diff
@@ -189,8 +189,48 @@ end
 
 --- @class vim.diagnostic.Opts.Status
 ---
---- A table mapping |diagnostic-severity| to the text to use for each severity section.
---- @field text? table<vim.diagnostic.Severity,string>
+--- Only show diagnostics matching the given severity |diagnostic-severity|.
+--- @field severity? vim.diagnostic.SeverityFilter
+---
+--- Either:
+--- - a table mapping |diagnostic-severity| to the text to use for each
+---   existing severity section. The default formatting will be applied.
+---   Default format: `E:1 W:2 I:3 H:4`, zeros omitted. Example:
+---   ```lua
+---   vim.diagnostic.config({
+---     status = { format = { [vim.diagnostic.severity.ERROR] = "󰅚" } }
+---   })
+---   ```
+---   With 1 error diagnostic associated with the buffer and 0 for other
+---   sections the status line will be: `󰅚:1`.
```

Comment:
I'm having trouble reading this. I don't think we need an example here.

```suggestion
--- - a table mapping |diagnostic-severity| to the text to use for each
---   existing severity section.
```